Framework/Kernel/src/Unit.cpp

```diff
@@ -644,56 +644,80 @@ void dSpacing::init() {
       }
     }
   }
+  // force the assumption that difc is positive in debug builds
+  assert(difc > 0.);
```

Comment:
why not throw a runtime error when encountering negative difc?   It is more informative

Reply:
`assert` only runs in debug builds and the vast majority of times people are going to give this positive values. When something is "wrong" a developer will generally switch to a debug build and this will kick in and tell them that an assumption has been violated without any (normal) runtime performance hit. The other runtime errors are more likely to happen.